homeassistant/components/aprilaire/sensor.py

```diff
@@ -0,0 +1,309 @@
+    ) -> None:
+        """Initialize a sensor for an Aprilaire device."""
+
+        self.entity_description = description
+
+        super().__init__(coordinator, unique_id)
+
+    @property
+    def exists(self) -> bool:
+        """Return True if the sensor exists."""
+
+        if (
+            self.entity_description.status_key is None
+            or self.status_sensor_exists_values is None
+        ):
+            return True
+
+        return (
+            self.coordinator.data.get(self.entity_description.status_key)
+            in self.status_sensor_exists_values
+        )
+
+    @property
+    def available(self) -> bool:
+        """Return True if the sensor is available."""
+
+        if (
+            self.entity_description.status_key is None
+            or self.status_sensor_available_value is None
+        ):
+            return True
+
+        if not super().available:
+            return False
+
+        return (
+            self.coordinator.data.get(self.entity_description.status_key)
+            == self.status_sensor_available_value
+        )
+
+    @property
+    def native_value(self) -> StateType:
+        """Return the value reported by the sensor."""
+
+        # Valid cast as pyaprilaire only provides str | int | float
+        return cast(
+            StateType, self.coordinator.data.get(self.entity_description.value_key)
+        )
+
+
+class AprilaireHumiditySensor(BaseAprilaireSensor):
+    """Humidity sensor entity for Aprilaire."""
+
+    status_sensor_available_value = 0
+    status_sensor_exists_values = [0, 1, 2]
+
+
+class AprilaireTemperatureSensor(BaseAprilaireSensor):
+    """Temperature sensor entity for Aprilaire."""
+
+    status_sensor_available_value = 0
+    status_sensor_exists_values = [0, 1, 2]
+
+    @property
+    def suggested_display_precision(self) -> int | None:
+        """Return the suggested number of decimal digits for display."""
+        if self.unit_of_measurement == UnitOfTemperature.CELSIUS:
+            return 1
+
+        return 0
+
+
+class AprilaireStatusSensor(BaseAprilaireSensor):
+    """Status sensor entity for Aprilaire."""
+
+    status_sensor_exists_values = [1, 2]
+    entity_description: AprilaireStatusSensorDescription
+
+    @property
+    def native_value(self) -> StateType:
+        """Return the value reported by the sensor mapped to the status option."""
+
+        raw_value = super().native_value
+
+        return self.entity_description.status_map.get(cast(int, raw_value))
+
+
+def get_entities(
+    entity_class: type[BaseAprilaireSensor],
+    coordinator: AprilaireCoordinator,
+    unique_id: str,
+    descriptions: tuple[AprilaireSensorDescription, ...],
+) -> list[BaseAprilaireSensor]:
+    """Get the entities for a list of sensor descriptions."""
+
+    entities = (
+        entity_class(coordinator, description, unique_id)
+        for description in descriptions
+    )
+
+    return [entity for entity in entities if entity.exists]
```

Comment:
make `BaseAprilaireSensor.exists` a class method so we don't instantiate objects only to immediately discard them

Reply:
@emontnemery The `exists` method is dependent on instance variables so it can't be a class method right now. It probably could be with some substantial refactoring but I don't think it's worth it.